### PR TITLE
fix: redirect to not found page if filter is invalid

### LINKED ISSUE #458: [Bug]: Fix filters bug
### What happened?

On entering an invalid filter, the user should be redirected to NotFound page.

### Add screenshots

Add screenshots

### What browsers are you seeing the problem on?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by kailashchoudhary11**
fixes #458

### CODE REVIEW

File: src/pages/Bookmarks.jsx [hunk 1]
```diff
@@ -2,15 +2,21 @@ import React, { useContext, useState, useEffect } from 'react'
...
-import { useSearchParams } from 'react-router-dom'
+import { useSearchParams, Navigate } from 'react-router-dom'
 import filterProducts from '../utils/filter/filter_products'
+import { allFilters } from '../utils/data/filters'
 
 const BookMarks = ({ length }) => {
   const { bookmarkfilteredProducts, deleteres } =
     useContext(ToolContext);
   const [searchParams] = useSearchParams()
   let filters = searchParams.getAll('filters').length > 0 ? searchParams.getAll('filters') : ["all"]
```
Comment (VenomFate-619) [inline src/pages/Bookmarks.jsx:13]: what's the use of getAll if you are using the filter's first index to search
`filters = filters[0].split(",")`

File: src/pages/Home.jsx [hunk 2]
```diff
@@ -17,7 +18,11 @@ const Card = ({ length }) => {
   const [searchParams,] = useSearchParams()
   let filters = searchParams.getAll('filters').length > 0 ? searchParams.getAll('filters') : ["all"]
```
Comment (VenomFate-619) [inline src/pages/Home.jsx:19]: what's the use of getAll if you are using the filter's first index to search
`filters = filters[0].split(",")`

**Comment by VenomFate-619** [comment]
the code is repetitive , do abstracted it into a function or a hook

**Comment by kailashchoudhary11** [comment]
> the code is repetitive, do abstract it into a function or a hook

@VenomFate-619 Done, Review it.

**Review by VenomFate-619 — APPROVED** [review]

**Pull request merged**